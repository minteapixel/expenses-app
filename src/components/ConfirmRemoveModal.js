import React from 'react';
import Modal from 'react-modal';

const ConfirmRemoveModal = (props) => (
	<Modal 
		isOpen = {props.showModal}
		onRequestClose={props.handleCloseModal}
		contentLabel="Confirm Removal Expense"
		closeTimeoutMS={200}
		className="modal"
	>
		<h3 className="modal__title">Confirm Expense Removal</h3>
		<p className="modal__body">
			Are you sure you want to remove the below expense?
		</p>
		
		<p className="modal__body modal__body--bold">
			Removing: <br /> {props.description}
		</p>

		<button className="login-button login-button--warning modal__button" onClick={props.onRemove}>
			Remove expense
		</button>

		<button className="login-button login-button__secondary modal__button" onClick={props.handleCloseModal}>
			Keep expense
		</button>
	</Modal>
)

if(process.env.NODE_ENV !== 'test') {
	Modal.setAppElement('#app');
}

export default ConfirmRemoveModal;