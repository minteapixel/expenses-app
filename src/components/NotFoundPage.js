import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
	<div>
		404! Oopsies! - Return to the homepage <Link to="/">here</Link>.
	</div>
);

export default NotFoundPage;