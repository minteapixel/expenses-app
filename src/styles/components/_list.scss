.list-header {
	background: $purple;
	border: 1px solid $purple;
	border-radius: 6px 6px 0 0;
	color: $white;
	display: flex;
	justify-content: space-between;
	padding: $s-size $m-size;
}

.list-body {
	margin-bottom: $m-size;
	@media (min-width: $desktop-breakpoint) {
		margin-bottom: $l-size;
	}
}

.list-item {
	border: 1px solid darken($light-grey, 10%);
	border-top: none;
	color: $dark-grey;
	display: flex;
	flex-direction: column;
	padding: $s-size;
	text-decoration: none;
	transition: background .3s ease;
	&:nth-child(even) {
		background-color: $light-grey;
	}
	&:hover {
		background: $light-orange;
	}
	@media (min-width: $desktop-breakpoint){
		align-items: center;
		flex-direction: row;
		justify-content: space-between;
		padding: $m-size;
	}
}

.list-item--message {
	color: $grey;
	align-items: center;
	justify-content: center;
	padding: $m-size;
	&:hover {
		background: none;
	}
}

.list-item__title {
	margin: 0;
	word-break: break-all;
}

.list-item__sub-title {
	color: $grey;
	font-size: $font-size-small;
}

.list-item__data {
	margin: $s-size 0 0 0;
	@media (min-width: $desktop-breakpoint){
		margin: 0;
		padding-left: $s-size;
	}
}