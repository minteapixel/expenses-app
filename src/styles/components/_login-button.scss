.login-button {
	background: $light-blue;
	border: none;
	color: $white;
	display: inline-block;
	font-weight: 300;
	font-size: $font-size-large;
	line-height: 1;
	padding: $s-size;
	text-align: center;
	text-decoration: none;
}

.login-button--link {
	background: none;
} 

.login-button--secondary {
	background: #888;
}

.login-button--warning {
	background: $orange;
	color: $white;
}