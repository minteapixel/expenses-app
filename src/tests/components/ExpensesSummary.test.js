import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render with 1 expense correctly', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render with multiple expenses correctly', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235123434} />);
	expect(wrapper).toMatchSnapshot();
});