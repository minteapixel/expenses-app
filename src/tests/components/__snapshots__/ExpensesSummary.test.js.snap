// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with 1 expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Dashboard
    </h1>
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.35
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="login-button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render with multiple expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Dashboard
    </h1>
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling 
      <span>
        $2,351,234.34
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="login-button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
